docs/configuration/CONFIGURATION.md

```diff
@@ -45,36 +45,176 @@ This overview is still a work in progress. You can refer to the [example.config.
 
 :::
 
-### URL
+### Application URL
 
-UniCore's base URL. This value is communicated to clients and identity wallets.
+UniCore's application URL. This value represents the self-aware URL of the application. It is used for internal
+communication and should not be exposed to clients or identity wallets.
 
-| Environment variable | `config.yaml` |
-| -------------------- | ------------- |
-| `UNICORE__URL`       | `url`         |
+:::note
+
+The `UNICORE__APPLICATION_URL` may include a path segment, which will be treated as the base path for the application. All endpoints will be attached relative to this base path. For example, if you set `UNICORE__APPLICATION_URL` to `http://localhost:3033/my/base/path`, then all API endpoints will be served under `/my/base/path`.
+
+:::
+
+| Environment variable       | `config.yaml`     |
+| -------------------------- | ----------------- |
+| `UNICORE__APPLICATION_URL` | `application_url` |
+
+#### Example
+
+```yaml
+url: http://localhost:3033/my/base/path
+```
+
+### Public URL
+
+UniCore's public URL. This value is communicated to clients and identity wallets and should be publicly accessible. When
+not set, it defaults to the value of `UNICORE__APPLICATION_URL`.
+
+<!-- TODO: In production, should we require that `UNICORE__PUBLIC_URL` is always explicitly set, rather than defaulting to the value of `UNICORE__APPLICATION_URL`? This would help prevent accidental exposure of internal URLs to clients. -->
+
+:::note
+
+The `UNICORE__PUBLIC_URL` may also include a path segment, which will be treated as the base path for all public endpoints. For example, if you set `UNICORE__PUBLIC_URL` to `https://my-domain.example.test/my/base/path`, then all public API endpoints will be served under `/my/base/path`.
+
+:::
+
+| Environment variable  | `config.yaml` |
+| --------------------- | ------------- |
+| `UNICORE__PUBLIC_URL` | `public_url`  |
 
 #### Example
 
 ```yaml
 url: https://my-domain.example.test
 ```
 
-### Port
+<!-- TODO: We should add a better explanation to describe the difference between the `UNICORE__APPLICATION_URL` and the `UNICORE__PUBLIC_URL`, possibly with some diagrams. Is this the right place for that? -->
 
-The port UniCore's HTTP server listens on.
+### Token Endpoint
 
-| Environment variable | `config.yaml` |
-| -------------------- | ------------- |
-| `UNICORE__PORT`      | `port`        |
+The OAuth2/OpenID Connect token endpoint. This endpoint is used by clients to exchange authorization codes for access tokens.
 
-#### Values
+:::note
+
+This variable is **optional**. By default, the `UNICORE__PUBLIC_URL` is used as the base, and the `/auth/token` segment is appended to form the token endpoint URL. You can completely override this default by explicitly setting the `UNICORE__TOKEN_ENDPOINT` variable or the `token_endpoint` config value.
+
+In most setups, the default value is recommended and usually the best choice.
```

Comment:
I suggest a different wording here: **For most setups, the default value is recommended and should only be changed if you know what you're doing.**

Reply:
Agree :+1: